transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/ProxyFactory.java

```diff
@@ -0,0 +1,86 @@
+package com.gruelbox.transactionoutbox;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.function.BiFunction;
+import net.bytebuddy.ByteBuddy;
+import net.bytebuddy.TypeCache;
+import net.bytebuddy.TypeCache.Sort;
+import net.bytebuddy.description.modifier.Visibility;
+import net.bytebuddy.dynamic.loading.ClassLoadingStrategy;
+import net.bytebuddy.implementation.InvocationHandlerAdapter;
+import net.bytebuddy.matcher.ElementMatchers;
+import org.objenesis.Objenesis;
+import org.objenesis.ObjenesisStd;
+import org.objenesis.instantiator.ObjectInstantiator;
+
+public class ProxyFactory {
+
+  private static final Objenesis objenesis = new ObjenesisStd();
+  private static final TypeCache<Class<?>> byteBuddyCache = new TypeCache<>(Sort.WEAK);
```

Comment:
Does this need to be static, based on the usage patterns you have in place?

Reply:
Good point, forgot to update this after stripping it out of `Utils`.

Reply:
This will fail on class init if bytebuddy isn't present.

Reply:
Updated to catch `NoClassDefFoundError` when initialising the object, and set these things as `null`. We then check for null before we use them.